backend/graphql/usecases/create-order.use-case.ts

```diff
@@ -0,0 +1,52 @@
+/* eslint-disable @typescript-eslint/no-non-null-assertion */
+import { OrderModel } from '../../database/models/order.model'
+import { ProductModel } from '../../database/models/product.model'
+import { NotFoundError } from '../errors/notFoundError'
+import { type CreateOrderInput } from '../inputs/order.input'
+import { type Context } from '../types/context.type'
+import { type Order } from '../types/order.type'
+
+export class CreateOrderUseCase {
+  async execute(input: CreateOrderInput, context: Context): Promise<Order> {
+    const user = context.user
+
+    if (!user) {
+      throw new NotFoundError('User not found')
+    }
+
+    const _id = input.items.map((item) => item.productId)
+
+    const products = await ProductModel.find({
+      _id
+    })
+
+    if (products.length !== input.items.length) {
+      throw new NotFoundError('Some products not found')
+    }
+
+    const orderItems = input.items.map((item) => {
+      const product = products.find(
+        (product) => product._id.toString() === item.productId
+      )
+
+      return {
+        productId: product!._id,
+        price: product!.price,
+        quantity: item.quantity
+      }
+    })
+
+    const totalPrice = orderItems
```

Comment:
this returns a string, shouldn't we store it as a number?

Reply:
you're right! but IDK why the result was right on the database. but I added Number() around it